src/osd/PG.cc

```diff
@@ -2646,12 +2653,29 @@ void PG::scrub(epoch_t queued, ThreadPool::TPHandle &handle)
  */
 void PG::chunky_scrub(ThreadPool::TPHandle &handle)
 {
+  // Since repair is only by request and we need to scrub afterward
+  // treat the same as req_scrub.
+  if (!scrubber.req_scrub && !scrubber.check_repair) {
```

Comment:
Do we need the `check_repair` check based on https://github.com/ceph/ceph/pull/35909#issuecomment-661539277?

Reply:
> Do we need the `check_repair` check based on [#35909 (comment)](https://github.com/ceph/ceph/pull/35909#issuecomment-661539277)?

It was discovered by "tasks/repair_test" teuthology test.  Since a direct repair does a check_repair (do a deep scrub) it should NOT be affected by noscrub/nodeep-scrub.  Of course, now that I think about it, maybe if nodeep-scrub is set after an auto repair already happened and it set check_repair that deep-scrub would NOT be aborted.  It could be argued that if you have come that far and completed an auto repair, you should go ahead and finish all the clean-up.  On the other hand the user may want all schedule scrubbing related activity to stop. immediately.